### ISSUE TITLE
Tests for Lua Routes

### ISSUE THREAD

**Issue by tobiasschweizer** [dhlab-basel/Sipi#245]
I saw that you moved the config files and tests for Knora Lua scripts into the Knora integration tests. That's very good since Sipi should not contain Knora specific things!

However, I do not know if the python tests are stilled used for Sipi:

https://github.com/dhlab-basel/Sipi/blob/f9f5380c6b7d31cfa0df7fc58bb504a00aa5df36/test/e2e/test_02_server.py#L37-L43

These tests cannot be run anymore because these routes do not exist in the Sipi repo anymore.

While `test_mediatype` is clearly Knora specific, `test_functions` could probably remain in the Sipi repo. What do you think?

**Comment by subotic** [comment]
They are still there, just in another place: https://github.com/dhlab-basel/Sipi/tree/develop/test/_test_data

Anything related to tests is now under the `test` folder. This still needs to be cleaned up. I suppose that over time the only python tests left, will be the official IIIF tests. Lukas is only adding native C++ tests and the few end-to-end tests written in python could/should be rewritten in C++.

**Comment by tobiasschweizer** [comment]
> I suppose that over time the only python tests left, will be the official IIIF tests.

I think it makes sense to have a test that just checks the Lua functions Sipi offers, but in a non Knora related way.

**Comment by subotic** [comment]
> I think it makes sense to have a test that just checks the Lua functions Sipi offers, but in a non Knora related way.

Yes, of course. I agree. Those we don't have at the moment and that's why I left the Knora tests so that we at least have something.

These kinds of tests should also be possible with the C++ native test framework. It should be possible to start `Sipi` with a given configuration from C++ and then run a few HTTP requests.

**Comment by subotic** [comment]
> So this would mean that in the test a Lua route would be called?

Yes, much like in the python tests now.
